### PR TITLE
[cleanup][pvr] get rid of obsolete guide views

### PR DESCRIPTION
**Pull request by xhaggi**
As discussed this PR will remove the different guide views "channel", "now" and "next".
Needs to be merged before #10795

@ksooo mind taking a look

### CODE REVIEW

**Comment by xhaggi** [comment]
not sure if we also want to drop the language keys `19030`, `19031`

**Comment by ronie** [comment]
> not sure if we also want to drop the language keys 19030, 19031

according to the language file, they are used for other stuff as well

File: xbmc/pvr/windows/GUIWindowPVRGuide.cpp [hunk 1]
```diff
@@ -207,71 +204,25 @@ void CGUIWindowPVRGuide::GetContextButtons(int itemNumber, CContextButtons &butt
...
 
 bool CGUIWindowPVRGuide::GetDirectory(const std::string &strDirectory, CFileItemList &items)
 {
-  switch (m_viewControl.GetCurrentControl())
-  {
-    case GUIDE_VIEW_TIMELINE:
-      GetViewTimelineItems(items);
-      break;
-    case GUIDE_VIEW_NOW:
-      GetViewNowItems(items);
```
Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: Declaration and definition of this member function should be not used anymore and can be dropped as well.

File: xbmc/pvr/windows/GUIWindowPVRGuide.cpp [hunk 2]
```diff
@@ -207,71 +204,25 @@ void CGUIWindowPVRGuide::GetContextButtons(int itemNumber, CContextButtons &butt
...
-      break;
-    case GUIDE_VIEW_NEXT:
-      GetViewNextItems(items);
```
Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: Declaration and definition of this member function should be not used anymore and can be dropped as well.

File: xbmc/pvr/windows/GUIWindowPVRGuide.cpp [hunk 3]
```diff
@@ -207,71 +204,25 @@ void CGUIWindowPVRGuide::GetContextButtons(int itemNumber, CContextButtons &butt
...
-      break;
-    case GUIDE_VIEW_CHANNEL:
-      GetViewChannelItems(items);
```
Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: Declaration and definition of this member function should be not used anymore and can be dropped as well.

File: xbmc/pvr/windows/GUIWindowPVRGuide.cpp [hunk 4]
```diff
@@ -288,9 +239,7 @@ bool CGUIWindowPVRGuide::OnAction(const CAction &action)
     case REMOTE_7:
     case REMOTE_8:
     case REMOTE_9:
-      if (m_viewControl.GetCurrentControl() != GUIDE_VIEW_CHANNEL)
         return ActionInputChannelNumber(action.GetID() - REMOTE_0);
```
Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: indentation!

**Review by ksooo — CHANGES_REQUESTED** [review]

**Comment by ksooo** [comment]
@xhaggi ping. have you seen my review comments? now that the master is open for v18 development i would like to merge this soon.

**Comment by xhaggi** [comment]
Yeah will address your comments in the next few days

**Comment by ksooo** [comment]
Please note that Estuary (Estouchy as well?) seems still to contain refereneces to the removed view types with numerical value 11, 12, 13. Please check and remove (grep output may contain false positives, but all guide and media menu related occurrences seem to be suspicious!).

<pre>
[kai@Kais-Mac-WLAN ~/src/github/ksooo/xbmc/addons/skin.estuary/xml (master)]$ grep -r "(11)" .
./DialogPVRChannelsOSD.xml:							<visible>Control.hasFocus(11)</visible>
./DialogPVRGuideOSD.xml:							<visible>Control.HasFocus(11)</visible>
./DialogPVRGuideOSD.xml:							<visible>!String.IsEmpty(ListItem.Plot) + Control.HasFocus(11)</visible>
./DialogPVRGuideOSD.xml:					<label>$LOCALIZE[19069]$INFO[Container(11).ListItem.StartDate, - ]</label>
./DialogVideoInfo.xml:						<param name="visible" value="System.hasAddon(script.extendedinfo) + String.IsEqual(ListItem.DBType,movie) + !Control.IsVisible(11)" />
./Includes_MediaMenu.xml:				<onright condition="Window.IsActive(MyPVRGuide.xml) + Control.IsVisible(11)">11</onright>
./Includes_MediaMenu.xml:						<onright condition="Window.IsActive(MyPVRGuide.xml) + Control.IsVisible(11)">11</onright>
./VideoOSDBookmarks.xml:				<visible>Integer.IsEqual(Container(11).NumItems,0)</visible>
./VideoOSDBookmarks.xml:							<visible>Control.HasFocus(11)</visible>
[kai@Kais-Mac-WLAN ~/src/github/ksooo/xbmc/addons/skin.estuary/xml (master)]$ grep -r "(12)" .
./Includes_MediaMenu.xml:				<onright condition="Window.IsActive(MyPVRGuide.xml) + Control.IsVisible(12)">12</onright>
./Includes_MediaMenu.xml:						<onright condition="Window.IsActive(MyPVRGuide.xml) + Control.IsVisible(12)">12</onright>
[kai@Kais-Mac-WLAN ~/src/github/ksooo/xbmc/addons/skin.estuary/xml (master)]$ grep -r "(13)" .
./DialogAddonInfo.xml:						<visible>Control.IsEnabled(13)</visible>
./DialogPVRGroupManager.xml:					<label>$LOCALIZE[31089]: [COLOR white]$INFO[Container(13).NumItems][/COLOR]</label>
./Includes_MediaMenu.xml:				<onright condition="Window.IsActive(MyPVRGuide.xml) + Control.IsVisible(13)">13</onright>
./Includes_MediaMenu.xml:						<onright condition="Window.IsActive(MyPVRGuide.xml) + Control.IsVisible(13)">13</onright>
[kai@Kais-Mac-WLAN ~/src/github/ksooo/xbmc/addons/skin.estuary/xml (master)]$ 
</pre>

**Comment by phil65** [comment]
Will take core of estuary cleanup.

**Comment by ksooo** [comment]
Thx @phil65

@xhaggi why must this PR go in before the horizontal epg grid PR? Cannot see a f dependency here.

**Comment by xhaggi** [comment]
Without you're unable to switch to the custom view type with horizontal epg.

**Comment by ronie** [comment]
i'll clean up estouchy once this is in.

**Comment by ksooo** [comment]
thx @ronie

**Comment by xhaggi** [comment]
@ksooo addressed your comments

**Review by ksooo — APPROVED** [review]

**Comment by ksooo** [comment]
thanks

jenkins build this please

**Comment by ksooo** [comment]
@xhaggi does not compile. please fix.

<pre>
/Users/Shared/jenkins/workspace/OSX-64/xbmc/pvr/windows/GUIWindowPVRGuide.cpp:235:9: error: non-void function 'GetDirectory' should return a value [-Wreturn-type]
        return;
</pre>

**Comment by xhaggi** [comment]
Force-pushed. Jenkins build this please

**Pull request merged**